protostar/commands/test/test_command.py

```diff
@@ -248,7 +260,9 @@ async def test(
                 slowest_tests_to_report_count=slowest_tests_to_report_count,
                 project_root_path=self._project_root_path,
             )
-            TestScheduler(live_logger=live_logger, worker=TestRunner.worker).run(
+            worker = PureCairoTestRunner.worker if pure_cairo_vm else TestRunner.worker
```

Comment:
This approach doesn't allow running StarkNet or Cairo Tests within a single run. The logic that decides what kind of test should be run needs to be deeper.